core/identity-hub-client/src/main/java/org/eclipse/edc/identityhub/client/IdentityHubClientImpl.java

```diff
@@ -124,16 +126,26 @@ public StatusResult<Collection<SignedJWT>> getVerifiableCredentials(String hubBa
     }
 
     @Override
-    public StatusResult<Void> addVerifiableCredential(String hubBaseUrl, SignedJWT verifiableCredential) {
-        var payload = verifiableCredential.serialize().getBytes(UTF_8);
+    public StatusResult<Void> addVerifiableCredential(String hubBaseUrl, CredentialEnvelope verifiableCredential) {
+
+        var transformer = transformerRegistry.resolve(verifiableCredential.format());
+        if (transformer == null) {
+            return StatusResult.failure(ResponseStatus.FATAL_ERROR, format("Transformer not found for format %s", verifiableCredential.format()));
+        }
+        Result<byte[]> result = transformer.serialize(verifiableCredential);
```

Comment:
`var`?

Reply:
It's needed for the compiler that it's not able to infer the right type and complains on `result.getContent()` below